src/main.rs

```diff
@@ -47,27 +47,41 @@ fn main() -> Result<(), JaktError> {
                     } else {
                         String::from("runtime")
                     };
-                    let default_cxx_compiler_path = PathBuf::from("clang++");
-                    let clang_output = Command::new(
-                        arguments
-                            .cxx_compiler_path
-                            .as_ref()
-                            .unwrap_or(&default_cxx_compiler_path),
-                    )
-                    .args([
-                        "-fcolor-diagnostics",
-                        "-std=c++20",
-                        // These warnings if enabled create loads of unnecessary noise:
-                        "-Wno-unqualified-std-cast-call",
-                        "-Wno-user-defined-literals",
-                        "-I",
-                        &runtime_path,
-                        &input_cpp,
-                        "-o",
-                        &output_executable,
-                    ])
-                    .output()?;
-                    io::stderr().write_all(&clang_output.stderr)?;
+                    let cxx_compiler = if let Some(ref path) = arguments.cxx_compiler_path {
+                        path.to_owned()
+                    } else {
+                        PathBuf::from(env!("CXX", "No supported C++ compiler was found!"))
```

Comment:
Another thing, will this error with `"No supported C++ compiler was found!"` if `CXX` is not set and `cxx_compiler_path` is not provided? 

The help states the compiler defaults to clang++ https://github.com/SerenityOS/jakt/blob/ecc5d6755f1b46ed2c1b7ca82cadc4d5d989e1c1/src/main.rs#L128 

I think this probably should stay the case (so you could first check `cxx_compiler_path` then the `CXX` env, otherwise default to `clang++`).

Reply:
You could maybe check `clang++` exists and error otherwise?

Reply:
I've changed this to use std::env::var("CXX").unwrap_or("clang++")